src/code/templates/egg-sort-game.js

```diff
@@ -29,6 +29,13 @@ export default class EggSortGame extends Component {
     onEggPlaced: PropTypes.func.isRequired
   }
 
+  state = {
+    selectedBaskets: [],
+    clickedBasket: null,
+    selectedEggIndex: null,
+    selectedEgg: null
+  }
+
```

Comment:
The link you gave describes initializing state in the constructor as the ES6 way and using property initializers as the ES7 way.